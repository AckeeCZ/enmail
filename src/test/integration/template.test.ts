import { compileLodashTemplate } from '../../index';

describe('Template', () => {
    it('Compile lodash template', () => {
        const template = `
            <a href="<%= url %>">#<%= id %> Invoice (<%= dateTimeFormat(receivedAt, 'Europe/Berlin') %>)</a>
            <span>Price: <strong><%= currencyFormat(price, 'EUR') %></strong></span>
        `;
        const compiled = compileLodashTemplate(template, {
            url: 'https://ackee.cz',
            receivedAt: new Date('2020-07-01T14:50:32Z'),
            price: 125.78,
            id: 202001,
        })();
        expect(compiled).toMatchSnapshot();
    });
});
