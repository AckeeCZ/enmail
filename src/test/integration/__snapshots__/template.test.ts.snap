// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template Compile lodash template 1`] = `
"            <a href=\\"https://ackee.cz\\">#202001 Invoice (Wed, 7/1/2020, 4:50 PM)</a>
            <span>Price: <strong>â‚¬125.78</strong></span>
        "
`;
